pkgs/vpp/default.nix

```diff
@@ -1,21 +1,54 @@
-{ stdenv, pkgs }:
+{ stdenv, pkgs, runtimeShell, python3 }:
 
 stdenv.mkDerivation rec {
   name = "vpp-${version}";
-  version = "git-2022-07-21";
+  version = "22.10-rc2";
   src = pkgs.fetchFromGitHub {
     owner = "FDio";
     repo = "vpp";
-    rev = "c8cd079a0004b75892a08c7cac9a48b39e24e580";
-    hash = "sha256-fx6J/g8BQYf93VGJLR+Hx/wu16Hg+ijkrRP89NnP9e0=";
+    rev = "v${version}"; #"61bae8a54d14899337b0d0a7ca9b9367f6321951";
+    hash = "sha256-3b+cnEjbReCg+svVD4DZwgcLwoA/IDWTGoP9ALYZFR4=";
   };
+  sourceRoot = "source/src";
 
-  nativeBuildInputs = with pkgs; [ pkg-config curl git cmake ninja nasm ];
+  nativeBuildInputs = with pkgs; [ pkg-config curl git cmake ninja nasm coreutils ];
   buildInputs = with pkgs; [
     libconfuse numactl libuuid
-    libffi libnl openssl
-    dpdk
+    libffi openssl
+
+    python3.pkgs.wrapPython(python3.withPackages (pp: with pp; [
+      ply
+    ]))
+
+    # linux-cp deps
+    libnl libmnl
+
+    # af_xdp deps
+    libbpf
+
+    # dpdk - needs static libraries, dunno how to do that.
+    #dpdk
   ];
-  buildPhase = "make build-release";
-  installPhase = "install -Dm755 example $out";
+
+  # Needs patches..
+  patchPhase = ''
+    # This attempts to use git to fetch the version, but we already know it.
+    printf "#!${runtimeShell}\necho '${version}~0-fakerev'\n" > scripts/version
```

Comment:
I might change this to actually show the revision information, but I tried to make it pull the actual git with tags and that failed horribly.

Reply:
Since it's a tagged release, i'll just set it to the tagged release.